internal/clients/http/gcp/gcp_client.go

```diff
@@ -0,0 +1,91 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/aws/smithy-go/ptr"
+	"github.com/rs/zerolog"
+)
+
+type gcpClient struct {
+	instancesClient *compute.InstancesClient
+	context         context.Context
+	logger          zerolog.Logger
+}
+
+func init() {
+	clients.GetGCPClient = newGCPClient
+}
+
+func newGCPClient(ctx context.Context) (clients.GCP, error) {
+	logger := ctxval.Logger(ctx).With().Str("client", "gcp").Logger()
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create new instances REST client: %w", err)
+	}
+
+	return &gcpClient{
+		instancesClient: instancesClient,
+		context:         ctx,
+		logger:          logger,
+	}, nil
+
+}
+
+func (c *gcpClient) RunInstances(ctx context.Context, projectID string, namePattern *string, imageName *string, DiskSizeGb int64, amount int64, machineType string, zone string, keyName string) error {
+	req := &computepb.BulkInsertInstanceRequest{
+		Project: projectID,
+		Zone:    zone,
+		BulkInsertInstanceResourceResource: &computepb.BulkInsertInstanceResource{
+			NamePattern: namePattern,
+			Count:       ptr.Int64(amount),
+			MinCount:    ptr.Int64(amount),
+			InstanceProperties: &computepb.InstanceProperties{
+				Disks: []*computepb.AttachedDisk{
+					{
+						InitializeParams: &computepb.AttachedDiskInitializeParams{
+							DiskSizeGb:  ptr.Int64(DiskSizeGb),
+							SourceImage: imageName,
+						},
+						AutoDelete: ptr.Bool(true),
+						Boot:       ptr.Bool(true),
+						Type:       ptr.String(computepb.AttachedDisk_PERSISTENT.String()),
+					},
+				},
+				MachineType: ptr.String(machineType),
+				NetworkInterfaces: []*computepb.NetworkInterface{
+					{
+						Name: ptr.String("global/networks/default"),
+					},
+				},
+				Metadata: &computepb.Metadata{
+					Items: []*computepb.Items{
+						{
+							Key:   ptr.String("ssh-keys"),
+							Value: ptr.String(keyName),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	c.logger.Trace().Msg("Executing Insert")
+	op, err := c.instancesClient.BulkInsert(ctx, req)
+	if err != nil {
+		return fmt.Errorf("cannot bulk insert instances: %w", err)
+	}
+
+	// TODO: info regarding the reservation and info about the created instances (create a db record for each instance)
+	if err = op.Wait(ctx); err != nil {
+		return fmt.Errorf("cannot wait for run instances operation to end: %w", err)
+	}
```

Comment:
I think you can drop waiting since you don't do anything after it finishes. We need to figure out how to get request/operation ID and store it in our database and how to do waiting later on. But for EC2 we also do not wait so let's just ignore this for now.